src/engraving/dom/edit.cpp

```diff
@@ -1327,43 +1327,42 @@ bool Score::rewriteMeasures(Measure* fm, const Fraction& ns, staff_idx_t staffId
 //    Add or change time signature at measure in response
 //    to gui command (drop timesig on measure or timesig)
 //---------------------------------------------------------
-
-void Score::cmdAddTimeSig(Measure* fm, staff_idx_t staffIdx, TimeSig* ts, bool local)
+void Score::cmdAddTimeSig(Measure* targetMeasure, staff_idx_t staffIdx, TimeSig* targetTimeSignature, bool local)
 {
     deselectAll();
 
-    if (fm->isMMRest()) {
-        fm = fm->mmRestFirst();
+    if (targetMeasure->isMMRest()) {
+        targetMeasure = targetMeasure->mmRestFirst();
     }
 
-    Fraction ns   = ts->sig();
-    Fraction tick = fm->tick();
-    TimeSig* lts  = staff(staffIdx)->timeSig(tick);
+    Fraction targetTimeSigFraction   = targetTimeSignature->sig();
```

Comment:
One more thing: as another cleanup measure, you could make all variables that don't need to change anymore `const`.

Reply:
I have added a bunch of consts, hopefully the type checking will stop anything breaking